tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,445 @@
+        self.poi = params.get("poi")
+        self.length = params.get("length")
+        self.height = params.get("height")
+
+    def exact_sol_global(self, sd: pp.Grid) -> np.ndarray:
+        """Compute the analytical solution for the pressurized fracture problem in
+        question.
+
+        Parameters:
+            sd: Subdomain for which the analytical solution is to be computed.
+
+        """
+
+        n = 1000
+        h = 2 * self.a / n
+        box_faces = sd.get_boundary_faces()
+        u_bc = np.zeros((sd.dim, sd.num_faces))
+
+        center = np.array([self.length / 2, self.height / 2, 0])
+        bem_centers = get_bem_centers(self.a, h, n, self.theta, center)
+        eta = compute_eta(bem_centers, center)
+
+        u_a = -analytical_displacements(
+            self.a, eta, self.p0, self.shear_modulus, self.poi
+        )
+        u_bc = assign_bem(sd, h / 2, box_faces, self.theta, bem_centers, u_a, self.poi)
+        return u_bc
+
+    def exact_sol_fracture(
+        self, mdg: pp.MixedDimensionalGrid
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Compute Sneddon's analytical solution for the pressurized crack
+        problem in question.
+
+        References:
+            Sneddon Fourier transforms 1951 page 425 eq 92
+
+        Parameters:
+            mdg: Mixed-dimensional domain of the setup.
+
+        Return:
+            A 2-tuple of numpy arrays, where the first vector contains the distances
+            between fracture center to each fracture coordinate, and the second vector
+            the respective analytical apertures.
+
+        """
+        ambient_dim = mdg.dim_max()
+        g_1 = mdg.subdomains(dim=ambient_dim - 1)[0]
+        fracture_center = np.array([self.length / 2, self.height / 2, 0])
+
+        fracture_faces = g_1.cell_centers
+
+        # compute distances from fracture centre with its corresponding apertures
+        eta = compute_eta(fracture_faces, fracture_center)
+        apertures = analytical_displacements(
+            self.a, eta, self.p0, self.shear_modulus, self.poi
+        )
+
+        return eta, apertures
+
+
+class ManuSneddonGeometry2d(pp.PorePyModel):
+    def set_domain(self) -> None:
+        """Defining a two-dimensional rectangle with parametrizable ``'length'`` and
+        ``'height'`` in the model parameters."""
+
+        self._domain = pp.Domain(
+            bounding_box={
+                "xmin": 0,
+                "ymin": 0,
+                "xmax": self.params["length"],
+                "ymax": self.params["height"],
+            }
+        )
+
+    def grid_type(self) -> Literal["simplex", "cartesian", "tensor_grid"]:
+        """Choosing the grid type for our domain."""
+        return self.params.get("grid_type", "simplex")
+
+    def set_fractures(self):
+        """Setting a single line fracture in the domain using ``params['frac_pts']``."""
+
+        points = self.params["frac_pts"]
```

Comment:
If the user provides a fracture which does not cross the domain center, the assumptions made in the analytical solution computation get invalidated. Should this be removed from the parameters and instead be hardcoded here?